### DISCUSSION TITLE
Vue3 - Error: No urql Client was provided

### DISCUSSION THREAD

**Discussion by pgrazek** [urql-graphql/urql#2582] [Q&A] [answered]
Hello, on a simple vue3 app, I'm trying to setup the client like this:
```
import { createApp } from 'vue'

import App from './App.vue'
import router from '@/router'
import { createClient } from '@urql/vue'

createApp(App)
    .use(router)
    .use(createClient({
        url: 'http://localhost:3000/api/graphql'
    }))
    .mount('#app')
```

and I'm getting this error : 

> useClient.ts:38 Uncaught Error: No urql Client was provided. Did you forget to install the plugin or call `provideClient` in a parent?

If I switch to .use() the urql import, it works just fine. What am I doing wrong here?

**Comment by kitten** [comment, marked as answer]
Because, technically, we don't expose a dedicated plugin object. We do expose `install` however: https://github.com/FormidableLabs/urql/blob/b4338ac7ae254a480450da68834630fb2a04d5d8/packages/vue-urql/src/useClient.ts#L16

So when you do `import * as urql from '@urql/vue'` then you get an object that contains the `install` property, since the package exports this, so it's the same as passing `app.use({ install }, opts)`

That's basically just down to what the Vue Plugin API expects when you use `app.use`: https://vuejs.org/guide/reusability/plugins.html

**Reply by pgrazek** [reply]
Thank you!
